ts/src/dna_factory.ts

```diff
@@ -325,13 +327,63 @@ export class DNAFactory {
       );
 
     const v = version ? unpad(version, this.encodingBase) : null;
-    if (!v) return this.generateNeftyDNALatest(archetypeIndex, dnaSchema, schemaVersion, rarityPreset);
+    if (!v) return this.generateNeftyDNALatest(archetypeIndex, grade, dnaSchema, schemaVersion, rarityPreset);
     if (v == '0' || v == '1') return this.generateNeftyDNAV0(archetypeIndex, dnaSchema, schemaVersion);
     else if (v == '2' || v == '3')
-      return this.generateNeftyDNALatest(archetypeIndex, dnaSchema, schemaVersion, rarityPreset);
+      return this.generateNeftyDNALatest(archetypeIndex, grade, dnaSchema, schemaVersion, rarityPreset);
     else throw new Error(`Invalid version ${version}`);
   }
 
+  generateStarterNeftyDNA(archetypeIndex: string) {
```

Comment:
Not really - I went for an extra method instead of adding another condition on generateNeftyDNA as the functionality seemed different enough (rarity and stats are both deterministic with starter).  Other standard eggs seemed have enough in common to share the same method though.